### PR TITLE
Moved cstring functions with other std includes

### LINKED ISSUE #6: io/ is a hidden dependency of boot/
boot/raspi3 and boot/linux-x86_64 use io/string.hh/cc for standard C string functions. This dependency shouldn't exist, as at the moment io is not automatically linked, and it's probably best to keep standard C string functions in libsupcxx, which is always linked.

### PR DESCRIPTION
**Pull request by ptia**
This closes #6, now boot/ only depends on the stdlib in libsupcxx/ and not on io/ .

### CODE REVIEW

File: libsupcxx/src/BUILD.go [hunk 1]
```diff
@@ -49,5 +49,7 @@ var Lib = lib.Library{
 		"eh_throw.cc",
 		"eh_catch.cc",
 		"eh_call.cc",
+
+		"cstring.cc",
```
Comment (jfrohnhofen) [inline libsupcxx/src/BUILD.go:53]: Also change the cmake files?

File: libsupcxx/src/cstring.cc [hunk 2]
```diff
@@ -37,10 +37,10 @@
 // SUCH DAMAGE.
 //------------------------------------------------------------------------------
 
-#include "libsupcxx/io/string.hh"
+#include <cstring>
+
 #include "limits.h"
 
-namespace {
```
Comment (jfrohnhofen) [inline libsupcxx/src/cstring.cc:43]: why did you remove this?

**Review by jfrohnhofen — APPROVED** [review]

File: libsupcxx/src/BUILD.go (hunk 1 above)
Comment (ptia) [reply libsupcxx/src/BUILD.go:53]: Yes, will do

File: libsupcxx/src/cstring.cc (hunk 2 above)
Comment (ptia) [reply libsupcxx/src/cstring.cc:43]: Yeah it should stay
